### PR TITLE
Jd 4 gh 39 store browser platform stats

### PR DESCRIPTION
**Pull request by geraltrivyan**

### CODE REVIEW

File: migrations/20200115142247_v1.10.0.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	});
+
+	await db.schema.createTable('country_platform_version_hits', (table) => {
+		table.increments();
+		table.string('countryIso', 2);
+		table.integer('platformVersionId').unsigned().references('id').inTable('platform_version').onUpdate('cascade').onDelete('cascade');
+		table.date('date').index();
+		table.integer('hits').unsigned().defaultTo(0).notNullable();
+		table.specificType('bandwidth', 'float').unsigned().defaultTo(0).notNullable();
+		table.unique([ 'platformVersionId', 'countryIso', 'date' ], 'unique_platformVersionId_country_date');
```
[the remark this answers is not in the document]
Comment (geraltrivyan) [reply migrations/20200115142247_v1.10.0.js]: Yes, generated one is too long

File: migrations/20200115142247_v1.10.0.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ exports.up = async (db) => {
 	await db.schema.createTable('platform_version', (table) => {
 		table.increments();
 		table.integer('platformId').unsigned().references('id').inTable('platform').onUpdate('cascade').onDelete('cascade');
-		table.string('version');
+		table.string('version').nullable();
```
Comment (MartinKolarik) [inline migrations/20200115142247_v1.10.0.js]: An empty string is not null.

File: migrations/20200115142247_v1.10.0.js.sql [hunk 3]
```diff
@@ -1,97 +1,97 @@
 drop function if exists updateOrInsertPlatform;
-create function updateOrInsertPlatform(aName varchar(255)) returns int
+create function updateOrInsertPlatform (aName varchar(255)) returns int
 begin
-	update `platform`
```
Comment (MartinKolarik) [inline migrations/20200115142247_v1.10.0.js.sql]: Formatting should be left as before this commit.

File: migrations/20200115142247_v1.10.0.js.sql [hunk 4]
```diff
@@ -0,0 +1,97 @@
+drop function if exists updateOrInsertPlatform;
+create function updateOrInsertPlatform (aName varchar(255)) returns int
+begin
+    update `platform`
```
Comment (MartinKolarik) [inline migrations/20200115142247_v1.10.0.js.sql]: Tabs.

File: src/models/CountryBrowserVersionHits.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const Joi = require('joi');
+const BaseModel = require('./BaseModel');
+
+const schema = {
+	countryIso: Joi.string().length(2).required(),
+	browserVersionId: Joi.string().length(2).required().allow(null),
```
Comment (MartinKolarik) [inline src/models/CountryBrowserVersionHits.js]: This looks wrong.

File: src/models/CountryPlatformVersionHits.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+const Joi = require('joi');
+const BaseModel = require('./BaseModel');
+
+const schema = {
+	countryIso: Joi.string().length(2).required(),
+	platformVersionId: Joi.string().length(2).required().allow(null),
```
Comment (MartinKolarik) [inline src/models/CountryPlatformVersionHits.js]: Same.

File: migrations/20200115142247_v1.10.0.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+const fs = require('fs');
+const sql = fs.readFileSync(__filename + '.sql', 'utf8');
+
+exports.up = async (db) => {
+	await db.schema.createTable('platform', (table) => {
+		table.increments();
+		table.string('name');
```
Comment (MartinKolarik) [inline migrations/20200115142247_v1.10.0.js]: .unique()

File: src/models/CountryBrowserVersionHits.js (hunk 5 above)
Comment (geraltrivyan) [reply src/models/CountryBrowserVersionHits.js]: Needed to pass validation, as long as `browserVersionId` we will get through SQL

Comment (geraltrivyan) [reply src/models/CountryBrowserVersionHits.js]: Oh your were talking about `length`... yeah, copy-pasta failed me

**Pull request merged**